pyproject.toml

```diff
@@ -70,7 +70,10 @@ nbconvert = "6.4.4"
 
 # Optional dependencies for the ZenServer
 fastapi = { version = "~0.75.0", optional = true }
-uvicorn = { extras = ["standard"], version = "~0.17.5" , optional = true}
+uvicorn = { extras = ["standard"], version = "~0.17.5",  optional = true}
+
+# dependencies for stack recipes
+python-terraform = "^0.10.1"
```

Comment:
This line makes the terraform python package an explicit dependency of ZenML, is this intended? I haven't checked how large this package is, but we could maybe think about having it optional as well?

Reply:
I would suggest we move it as an extra so `pip install zenml[terraform]` or something?

Reply:
If we do the above, we should add some logic that tries to import this in the __init__ of stack recipes and fails gracefully if not installed

Reply:
Yes I agree, `zenml stack recipe` could simply print a warning/info message that it's missing this package.

```python
try:
  import python_terraform
except ImportError:
  # add just the stack recipe group here with info on how to install it/make it available
else:
  # the normal stack recipe CLI code here
```

Reply:
That makes sense! Should I go with `zenml[recipes]`  or `zenml[terraform]` for the extras installation?

Reply:
`zenml[recipes]` might make sense